custom_components/echonetlite/sensor.py

```diff
@@ -26,35 +33,46 @@ async def async_setup_entry(hass, config, async_add_entities, discovery_info=Non
         if eojgc == 1 and eojcc == 48: 
             for op_code in ENL_SENSOR_OP_CODES[eojgc][eojcc].keys():
                 if op_code in entity['instance']['getmap']:
-                    entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES[eojgc][eojcc][op_code], hass.config.units, config.title))
+                    entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES[eojgc][eojcc][op_code], config.title))
         else: #handle other ECHONET instances
             for op_code in EPC_CODE[eojgc][eojcc]:
                 if eojgc in ENL_SENSOR_OP_CODES.keys():
                     if eojcc in ENL_SENSOR_OP_CODES[eojgc].keys():
                         if op_code in ENL_SENSOR_OP_CODES[eojgc][eojcc].keys():
-                            entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES[eojgc][eojcc][op_code], hass.config.units, config.title))
+                            entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES[eojgc][eojcc][op_code], config.title))
                         else:
-                            entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES['default'], hass.config.units, config.title))
+                            entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES['default'], config.title))
                 elif op_code in list(entity['echonetlite']._update_flags):
-                    entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES['default'], hass.config.units, config.title))
+                    entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES['default'], config.title))
     async_add_entities(entities, True)
 
 
-class EchonetSensor(Entity):
+class EchonetSensor(SensorEntity):
     """Representation of an ECHONETLite Temperature Sensor."""
 
-    def __init__(self, instance, op_code, attributes, units: UnitSystem, name=None) -> None:
```

Comment:
I've removed units from this sensor as echonet is always Celsius AFAIK so no need for this to be a parameter

Reply:
The sensor can be used by lots of things not just temperature values so I don’t really think this change is a good idea

Reply:
I saw a comment from you about this but it's disappeared.  The idea with the native values thing is to never do conversions based on the users configured unit systems - always pass the raw values the system gives you and let HA do any conversions within the sensor platform.